### PR TITLE
Add 8.1.0 release notes

### PR DESCRIPTION
**Pull request by idegtiarenko**
https://elasticsearch_83741.docs-preview.app.elstc.co/diff

### CODE REVIEW

File: docs/reference/release-notes.asciidoc
Comment (idegtiarenko) [inline docs/reference/release-notes.asciidoc:20]: @pugnascotia this was done automatically by `./gradlew generateReleaseNotes`, I do not think this is intended

**Review by jrodewig — COMMENTED** [review]
It looks like this PR accidentally removes the 8.0.0 release notes. We'll want to leave those in.

Otherwise, I just left some suggestions for minor, non-blocking nits.

**Review by kirakoki — APPROVED** [review]

**Comment by lcawl** [comment]
FYI I've integrated the PRs from https://github.com/elastic/ml-cpp/blob/8.1/docs/CHANGELOG.asciidoc

File: docs/reference/release-notes/8.1.asciidoc
Comment (lcawl) [inline docs/reference/release-notes/8.1.asciidoc:61]:
```suggestion
* Wait for model process to stop in stop deployment {es-pull}83644[#83644]
```

**Review by lcawl — APPROVED** [review]
One suggestion, otherwise ML PRs LGTM

File: docs/reference/release-notes/8.1.asciidoc
Comment (pugnascotia) [inline docs/reference/release-notes/8.1.asciidoc:208]: Can we make sure all the "Upgrade" items in this file go under the "Upgrades" section at the end? There are 3 ML items and 1 Security item to move.

**Review by pugnascotia — COMMENTED** [review]
It can happy after this PR is merged, but can we get someone from the ES Security team to check that security PRs in the notes are supposed to be there? Some of them may be for work-in-progress 🙏

**Review by pugnascotia — APPROVED** [review]

**Comment by ywangd** [comment]
> It can happy after this PR is merged, but can we get someone from the ES Security team to check that security PRs in the notes are supposed to be there? Some of them may be for work-in-progress 🙏

Thanks for the ping. Yes there are a few in-progress work that should not be in the 8.1 release notes:
* Activate user profile API [#82400]
* Add an initial `ProfileService` for user profiles [#81899]
* Add new system index for user profile documents [#81355]
* Add update user profile data API [#82772]
* Add user profile API for get profile by UID [#81910]
* Update Kibana system user privileges [#82781]
* Initial version of JWT Realm [#82175]
* Introduce domain setting to associate realms [#81968]

**Pull request merged**
